pkg/server/serverpb/status.proto

```diff
@@ -599,24 +599,15 @@ message GossipRequest {
   bool redact = 2;
 }
 
-message EngineStatsInfo {
-  int32 store_id = 1 [
-    (gogoproto.customname) = "StoreID",
-    (gogoproto.casttype) =
-        "github.com/cockroachdb/cockroach/pkg/roachpb.StoreID"
-  ];
-  cockroach.storage.enginepb.TickersAndHistograms tickers_and_histograms = 2;
-  cockroach.storage.enginepb.EngineType engine_type = 3;
-}
-
 message EngineStatsRequest {
   // node_id is a string so that "local" can be used to specify that no
   // forwarding is necessary.
   string node_id = 1;
 }
 
 message EngineStatsResponse {
-  repeated EngineStatsInfo stats = 1 [ (gogoproto.nullable) = false ];
+  // contains LSM metrics for all stores on the requested node
+  string metrics = 1;
```

Comment:
if we're changing the protobuf anyway, I'd have a slight preference for `repeated string metrics [(gogoproto.nullable=false)]`, that way we have more infrastructure ready for per-store responses if we ever move in that direction. You can move the concatenation logic into a helper that can then be used by `debug.GetEngineStats`.

Also convention with changing protobufs is that the new field doens't use the same argument ID (the `= 1` at the end) as the old field. You might need to make this 2, and you might also need a `reserved 1` above it for the missing 1, but maybe that last part isn't necessary.

Reply:
Thanks for calling out the argument ID convention - I've updated to use `= 2`.

I also opted to use a [store ID -> string map](https://github.com/cockroachdb/cockroach/pull/125865/files#diff-bdf84dc546a641c13d616508d4f49a4fdef55eaa1ee8a1eccb38114e3b37777dR610) rather than a list of strings so that we can filter by the store ID if needed in the future.